articles/openshift/oshift-netpol.md

```diff
@@ -0,0 +1,138 @@
+---
+title: How to use NetworkPolicy objects to connect services between projects | UKCloud Ltd
+description: Shows how to create a basic NetworkPolicy object to connect services in two projects
+services: openshift
+author: Kieran O'Neill
+
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Using network policy
+toc_fullpath: How To/oshift-netpol.md
+toc_mdlink: oshift-netpol.md
+---
+
+# How to use NetworkPolicy objects to connect services between projects
+
+## Overview
+
+In v3.11 clusters, deployed from mid-February 2019, the default SDN plugin has been changed from ovs-multitenant to ovs-networkpolicy. This gives you the ability to create NetworkPolicy objects which allow granular control over the flow of communication between pods/services/projects inside your cluster. 
+
+By default we put in place NetworkPolicy objects that allow all pods to communicate with each other inside the same project and allow communication from all pods/services to the default and openshift-monitoring projects. This mirrors the default pod security provided with the ovs-multitenant plugin. One of these objects will exist in all projects, including projects created after the cluster is deployed and have the name 'allow-from-same-and-privileged-namespaces'
+
+## Interacting with NetworkPolicy objects
+
+NetworkPolicy objects can be interacted with on the command line using the full name networkpolicy or short name netpol, for example:
+
+``` oc get netpol ``` - List NetworkPolicy objects in the current project.
+
+``` oc describe netpol <name> ``` - Provides more detailed information on the NetworkPolicy object.
+
+``` oc get netpol <name> -o yaml ``` - Provides yaml output of the NetworkPolicy object. Useful to reference existing rules when creating new objects.
+
+## Example of connecting services from two different projects
+
+NetworkPolicy objects are scoped at the project level. This means if you create a NetworkPolicy object using a template it will be created inside the project you're in and the rules will apply to pods within that project, with the exception of remote projects you may be referencing inside the rule. The object without any rules will look like the following:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: <desired_name_of_object>
+  namespace: <desired_project> (if not provided this will default to the project you are in when you run oc create)
+spec:
+```
+
+Inside the spec field we'll pass in the rules that will determine what our NetworkPolicy object does. The first thing you need to consider is the pods you want to target, this is done using the podSelector field. 
+You can use this to match all pods in a project or alternatively to match all pods with a certain label. If you leave this field blank it will match all pods. If you want to match pods with the label 'role=webserver' 
```

Comment:
```suggestion
You can use this to match all pods in a project or alternatively to match all pods with a certain label. If you leave this key blank it will match all pods. If you want to match pods with the label 'role=webserver' 
```